DBZ-6721 Add docs for SMTs (#5634)
debezium/debezium: documentation/modules/ROOT/pages/transformations/vitess-filter-transaction-topic-records.adoc

:page-aliases: configuration/vitess-filter-transaction-topic-records.adoc
[id="vitess-filter-transaction-topic-records"]
= Vitess Filter Transaction Topic Records

:toc:
:toc-placement: macro
:linkattrs:
:icons: font
:source-highlighter: highlight.js

toc::[]

When the Vitess connector is configured to provide transaction metadata, use the `FilterTransactionTopicRecords` SMT to prevent the connector from sending transaction boundary events to transaction topics.
This transformation is designed for use only with the {prodname} connector for Vitess.

To filter events, the SMT checks the key and value schemas of each record match and compares them to the values returned by `TransactionStructMaker`.

> The following example shows how to add the SMT to the connector configuration so that it filters out all transaction boundary event records that the connector emits so that they are not saved to the transacton topic.

Reviewer: ```suggestion
The following example shows how to add the `FilterTransactionTopicRecords` SMT to the connector configuration to remove all transaction boundary event records that the connector emits, and prevent them from being saved to the transaction topic.
```
